airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py

```diff
@@ -1233,6 +1252,22 @@ async def test_all_supported_file_types_present(self, certified_file_based_conne
             "or add them to the `file_types -> unsupported_types` list in config."
         )
 
+    @staticmethod
+    def _validate_stream_statuses(configured_catalog: ConfiguredAirbyteCatalog, statuses: List[AirbyteStreamStatusTraceMessage]):
+        """Validate all statuses for all streams in the catalogs were emitted in correct order"""
+        stream_statuses = defaultdict(list)
+        for status in statuses:
+            stream_statuses[status.stream_descriptor.name].append(status.status)
+
+        assert set(x.stream.name for x in configured_catalog.streams) == set(stream_statuses), "All stream must emit status"
+
+        for stream_name, status_list in stream_statuses.items():
+            assert status_list == [
+                AirbyteStreamStatus.STARTED,
+                AirbyteStreamStatus.RUNNING,
```

Comment:
Is there something that prevents connectors from sending multiple `AirbyteStreamStatus.RUNNING`? I would assume this could be useful in some case. I vaguely remember a case (I think it was source-stripe `bank_accounts`) where we needed to fetch multiple parents before finding a child with records and the platform would kill the stream after 60 minutes. I would assume that one of the solution we could take is to send `AirbyteStreamStatus.RUNNING` periodically when we know we didn't crash

Reply:
Hmm, sounds interesting, changed asserts to cover this case.